Policy engine based on JSON schema validation

johnandersen777 opened the pull request.

pdxjohnny committed:
docs: registration_policies: Leverage jsonschema for evaluation of claim

Signed-off-by: John Andersen <johnandersenpdx@gmail.com>

pdxjohnny committed:
docs: registration_policies: Decode COSE messages payload before JSON load

Signed-off-by: John Andersen <johnandersenpdx@gmail.com>

pdxjohnny committed:
Add jsonschema to dev-requirements.txt

Signed-off-by: John Andersen <johnandersenpdx@gmail.com>

pdxjohnny committed:
docs: registration_policies: Add policy_engine.sh wrapper and source policy from service parameters insertPolicy

Signed-off-by: John Andersen <johnandersenpdx@gmail.com>

pdxjohnny committed:
Add jsonschema to envrionment.yml

Signed-off-by: John Andersen <johnandersenpdx@gmail.com>

OR13 reviewed (commented):
An allow list would probably be a better example to start with, but overall looks good

pdxjohnny committed:
docs: registration_polices: Convert example from blocklist to allowlist per @OR13 review

Related: https://github.com/pdxjohnny/scitt-api-emulator/pull/1#pullrequestreview-1506375585
Signed-off-by: John Andersen <johnandersenpdx@gmail.com>
